File: Search a 2D Matrix/Q1.py
def createMatriz(n : int, m : int):
    matrix = []
    for i in range(0,n):
        array = []
        for j in range(0,m):
            x = int(input())
            array.append(x)
        matrix.append(array)
    
    return matrix

def binarySearch(list, target):
    l = 0
    r = len(list)
    while(l <= r):
        m = (l+r)//2
        if list[m] == target:
            return True
        if list[m] > target:
            r = m - 1
        else:
            l = m+1
    return False



n,m,target = map(int, input().split())

matrix = createMatriz(n,m)
flag = False
for i in matrix:
    flag = binarySearch(i,target)
    if flag:
        break

if flag:
    print('true')
else:
    print('false')

File: Search a 2D Matrix/Q1.cpp
#include<bits/stdc++.h>
using namespace std;

vector<vector<int>>M;

bool binary_search(vector<int> &a, int &target){
    int l = 0;
    int r = a.size() - 1;

    while(l <= r){
        int m = (l+r)/2;
        if(a[m] == target){
            return true;
        }

        if(a[m] > target){
            r = m-1;
        }else{
            l = m+1;
        }
    }
    return false;
}

int main(){

    int n, m, target;
    vector<vector<int>>M;
    cin >> n >>  m >> target;
    M.resize(n);
    for(int i = 0; i<n; i++){
        for(int j = 0; j<m; j++){
            int x; cin >> x;
            M[i].push_back(x);
        }
    }

    bool flag = false;
    for(int i = 0; i<n; i++){
        flag = binary_search(M[i], target);
        if( flag) break;
    }

    if(flag) cout << "true\n";
    else cout << "false\n";
    return 0;
}

